### PR TITLE
[gradle] gradle builder plugin doesn't support configuration of any j…

### LINKED ISSUE #3551: [gradle] gradle builder plugin doesn't support configuration of any jar tasks besides ones called exactly `jar`
For example, if someone applies the plugin:
```
	id("com.github.johnrengelman.shadow")
```
the `shadowJar` task cannot be configured to emit an OSGi bundle.

### PR DESCRIPTION
**Pull request by rotty3000**
…ar tasks besides ones called exactly `jar`

fixes #3551

Signed-off-by: Raymond Augé <raymond.auge@liferay.com>

### CODE REVIEW

File: biz.aQute.bnd.gradle/src/aQute/bnd/gradle/BndBuilderPlugin.groovy [hunk 1]
```diff
@@ -72,7 +73,7 @@ public class BndBuilderPlugin implements Plugin<Project> {
       tasks.register('baseline', Baseline.class) { t ->
         t.description 'Baseline the project bundle.'
         t.group 'release'
-        t.bundle jar
+        t.bundle jars.named('jar')
```
Comment (bjhargrave) [inline biz.aQute.bnd.gradle/src/aQute/bnd/gradle/BndBuilderPlugin.groovy]: This is wrong.

**Comment by rotty3000** [comment]
k, that was emphatic.

**Comment by bjhargrave** [comment]
> k, that was emphatic.

Sorry, perhaps I was in a cranky mood 🤭

Perhaps we can update the readme to provide code examples of applying the task convention to existing jar tasks?

```groovy
tasks.withType(Jar.class).configureEach { t ->
    t.convention.plugins.bundle = new BundleTaskConvention(t)
    t.doLast {
        buildBundle()
    }
}
```

Kotlin example left as an exercise for the reader :-)

**Comment by rotty3000** [comment]
I was able to do as you suggested. so the PR was bad, I just didn't realize how this was really supposed to be done.

**Pull request closed without merging**